extension/extension.go

```diff
@@ -43,6 +44,10 @@ type PipelineWatcher interface {
 	NotReady() error
 }
 
+type ConfigWatcher interface {
+	ConfigResolved(ctx context.Context, conf *confmap.Conf) error
```

Comment:
Is not "ConfigResolved" since that happened before the extension was created, not sure what a good name is...

Reply:
I've updated the method name to `NotifyConfig`.